inlineplz/linters/__init__.py

```diff
@@ -55,27 +59,703 @@ def vendored_path(path):
     )
 
 
-# glob patterns for what language is represented by what type of files.
-PATTERNS = {
-    "all": ["*.*"],
-    "ansible": ["*.yaml", "*.yml"],
-    "docker": ["*Dockerfile", "*.dockerfile"],
-    "gherkin": ["*.feature"],
-    "go": ["*.go"],
-    "groovy": ["*.groovy", "Jenkinsfile", "jenkinsfile"],
-    "java": ["*.java"],
-    "javascript": ["*.js"],
-    "json": ["*.json"],
-    "html": ["*.html", "*.htm"],
-    "markdown": ["*.md"],
-    "python": ["*.py"],
-    "shell": ["*.sh", "*.zsh", "*.ksh", "*.bsh", "*.csh", "*.bash"],
-    "stylus": ["*.styl"],
-    "robotframework": ["*.robot"],
-    "rst": ["*.rst"],
-    "text": ["*.md", "*.txt", "*.rtf", "*.html", "*.tex", "*.markdown"],
-    "yaml": ["*.yaml", "*.yml"],
-}
+GROOVY_PATH = vendored_path(os.path.join("groovy", "groovy-all-2.4.15.jar"))
+SLF4J_PATH = vendored_path(os.path.join("groovy", "slf4j-api-1.7.25.jar"))
+
+
+def register_patterns():
+    register_pattern("all", ["*.*"])
+    register_pattern("ansible", ["*.yaml", "*.yml"])
+    register_pattern("docker", ["*Dockerfile", "*.dockerfile"])
+    register_pattern("gherkin", ["*.feature"])
+    register_pattern("go", ["*.go"])
+    register_pattern("groovy", ["*.groovy", "Jenkinsfile", "jenkinsfile"])
+    register_pattern("java", ["*.java"])
+    register_pattern("javascript", ["*.js"])
+    register_pattern("json", ["*.json"])
+    register_pattern("html", ["*.html", "*.htm"])
+    register_pattern("markdown", ["*.md"])
+    register_pattern("python", ["*.py"])
+    register_pattern("shell", ["*.sh", "*.zsh", "*.ksh", "*.bsh", "*.csh", "*.bash"])
+    register_pattern("stylus", ["*.styl"])
+    register_pattern("robotframework", ["*.robot"])
+    register_pattern("rst", ["*.rst"])
+    register_pattern(
+        "text", ["*.md", "*.txt", "*.rtf", "*.html", "*.tex", "*.markdown"]
+    )
+    register_pattern("yaml", ["*.yaml", "*.yml"])
+
+
+def register_linters():
```

Comment:
I'm slightly tempted to leave these as constants that we `.format()`, but don't feel as strongly about these as the patterns.